src/main/js/dashboard-frontend/src/communication/ServerEndpoint.tsx

```diff
@@ -271,6 +295,37 @@ export default class ServerEndpoint {
         }
         break;
       }
+      case APICall.subscribeToPubSubTopic: {
+        const subId = request.args[0].subId;
+        const pot = this.pubSubTopicsSubscribers.get(subId);
+        if (pot === undefined || pot.size === 0) {
+          return this.sendRequest({...request, args: [JSON.stringify(request.args[0])]}).then((r) => {
```

Comment:
because we don't want to wait for the result here, the caller waits. And it also races against a 5 second timeout

Reply:
You can still `await` it won't wait. It is just syntactical sugar to do the same thing. It would execute and return a promise and only wait if the caller awaits.

```js
const expensiveOp = () => {
    const p = new Promise((res, rej) => {
        setTimeout(() => {
            res("Done")
        }, 5000)
    })

    return p;
}

const test = async (type: "test" | "test2") => {
    switch (type) {
        case "test":
            const res = await expensiveOp()

            if (typeof res === "string") {
                console.log("READY")
            } 

            return res;
        case "test2":
            return "Not expensive"
        default:
            throw new Error("Meh")
    }
}

test("test")
console.log("HERE")
``